Sources/Overload/OvEditor/src/OvEditor/Panels/MenuBar.cpp

```diff
@@ -197,8 +200,78 @@ void OvEditor::Panels::MenuBar::CreateSettingsMenu()
 
 void OvEditor::Panels::MenuBar::CreateLayoutMenu() 
 {
-	auto& layoutMenu = CreateWidget<MenuList>("Layout");
-	layoutMenu.CreateWidget<MenuItem>("Reset").ClickedEvent += EDITOR_BIND(ResetLayout);
+	auto& layoutMenuList = CreateWidget<MenuList>("Layout");
+
+	auto& uiManager = *EDITOR_CONTEXT(uiManager);
+	const std::filesystem::path& layoutsPath = uiManager.GetLayoutsPath();
+
+	auto& saveMenuItem = layoutMenuList.CreateWidget<MenuItem>("Save");
+	saveMenuItem.ClickedEvent += EDITOR_BIND(SaveCurrentLayout);
+
+	auto& saveNewMenuList = layoutMenuList.CreateWidget<MenuList>("Save New");
+	auto& layoutInputText = saveNewMenuList.CreateWidget<InputFields::InputText>("Layout Name");
+	layoutInputText.selectAllOnClick = true;
+	layoutInputText.EnterPressedEvent += [&layoutsPath](std::string p_input)
+	{
+		if (p_input.empty())
+			return;
+
+		auto& uiManager = *EDITOR_CONTEXT(uiManager);
+		EDITOR_EXEC(DelayAction(std::bind(&OvUI::Core::UIManager::SaveLayout, &uiManager, layoutsPath / (p_input + ".ini")), 1));
+	};
+
+	auto& loadMenuList = layoutMenuList.CreateWidget<MenuList>("Load");
+
+	loadMenuList.ClickedEvent += [&]
+	{
+		loadMenuList.RemoveAllWidgets();
+
+		for (const auto& entry : std::filesystem::directory_iterator(layoutsPath))
+		{
+			if (entry.is_regular_file() && entry.path().extension() == ".ini")
+			{
+				auto& layoutMenuItem = loadMenuList.CreateWidget<MenuItem>(entry.path().stem().string());
+				layoutMenuItem.name = entry.path().stem().string();
+
+				layoutMenuItem.ClickedEvent += [entry]
+				{
+					auto& uiManager = *EDITOR_CONTEXT(uiManager);
+					EDITOR_EXEC(DelayAction(std::bind(&OvUI::Core::UIManager::SetLayout, &uiManager, entry.path()), 1));
+				};
+				
+				auto& contextualMenu = layoutMenuItem.AddPlugin<OvUI::Plugins::ContextualMenu>();
+				auto& deleteMenuItem = contextualMenu.CreateWidget<MenuItem>("Delete");
+
+				deleteMenuItem.ClickedEvent += [entry, &layoutMenuItem]
+				{
+					auto& uiManager = *EDITOR_CONTEXT(uiManager);
+					EDITOR_EXEC(DelayAction(std::bind(&OvUI::Core::UIManager::DeleteLayout, &uiManager, entry.path()), 1));
+					layoutMenuItem.enabled = false;
+
+				};
+				auto& renameToMenuList = contextualMenu.CreateWidget<MenuList>("Rename to...");
+
+				auto& renameInputText = renameToMenuList.CreateWidget<InputFields::InputText>("");
+				renameInputText.content = entry.path().stem().string();
+				renameInputText.selectAllOnClick = true;
+
+				renameInputText.EnterPressedEvent += [entry, &layoutMenuItem, &layoutsPath, &contextualMenu](std::string p_newName)
+				{
+					if (p_newName.empty())
+						return;
+
+					layoutMenuItem.name = p_newName;
+					
+					auto& uiManager = *EDITOR_CONTEXT(uiManager);
+					EDITOR_EXEC(DelayAction(std::bind(&OvUI::Core::UIManager::RenameLayout, &uiManager, entry, layoutsPath / (p_newName + ".ini")), 1));
```

Comment:
Crashes when performed multiple times without closing the containing menu list.

https://github.com/user-attachments/assets/e44f7792-1915-41f6-89d3-3e9bbc269139

Reply:
Also, seems like the action delay isn't necessary. Got it to work with:
```cpp
uiManager.RenameLayout(entry, layoutsPath / (p_newName + ".ini"));
```

Reply:
Yep we are iterating other the entries resulting in not updated path when we pass the first parameter to RenameLayout

Reply:
Fixed, I now remember why I was using the `shared_ptr` before

Reply:
> Also, seems like the action delay isn't necessary. Got it to work with:
> 
> ```c++
> uiManager.RenameLayout(entry, layoutsPath / (p_newName + ".ini"));
> ```
> 
> Wondering if other actions also need to be delayed (`DeleteLayout`, `SetLayout`...)

`ResetToDefaultLayout` has to be delayed, otherwise, it results in un-docked panels with random positions due to the UI structure changing at the same time it is being rendered. 
I suggest keeping, at minimum, all methods that encapsulate `LoadIniSettingsFromDisk` in a delayed call.